frontend/src/scenes/funnels/Funnel.tsx

```diff
@@ -22,17 +25,27 @@ export function Funnel(props: Omit<ChartParams, 'view'>): JSX.Element | null {
     if (!areFiltersValid) {
         return <FunnelInvalidFiltersEmptyState />
     }
+    if (isLoading) {
+        return <Loading />
+    }
+    if (!isValidFunnel) {
+        return <FunnelEmptyState />
+    }
 
     if (featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ]) {
-        const funnel_viz_type = filters.funnel_viz_type || props.filters.funnel_viz_type
+        if (funnel_viz_type == FunnelVizType.Steps || !funnel_viz_type) {
+            return <FunnelBarGraph {...props} />
+        }
 
         if (funnel_viz_type == FunnelVizType.TimeToConvert) {
             return <FunnelHistogram {...props} />
         }
-        if (funnel_viz_type == FunnelVizType.Steps || !funnel_viz_type) {
-            return <FunnelBarGraph {...props} />
-        }
     }
 
+    if (funnel_viz_type === FunnelVizType.Trends) {
+        return <FunnelLineGraph {...props} />
+    }
+
+    // TODO: Remove this line when #4785 (Nail Funnels) has been rolled out to all users
```

Comment:
@paolodamico This is the only instance of `FunnelViz` that needs to be removed once you deprecate it completely. You'll also want to remove the `if(featureFlags[FEATURE_FLAGS.FUNNEL_BAR_VIZ])` wrapper above.